### PR TITLE
Terraform 3.19 updates

### PR DESCRIPTION
**Pull request by sebader**
Update provider and make use of new features

### CODE REVIEW

**Review by heoelri — APPROVED** [review]

**Comment by heoelri** [comment]
TF azurerm 3.19 was already superseeded by 3.19.1 which contains some dns resource related fixes. https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.19.1

File: src/infra/workload/globalresources/main.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.17.0"
+      version = "3.19.0"
```
Comment (sebader) [inline src/infra/workload/globalresources/main.tf:5]:
```suggestion
      version = "3.19.1"
```

File: src/infra/workload/releaseunit/main.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.17.0"
+      version = "3.19.0"
```
Comment (sebader) [inline src/infra/workload/releaseunit/main.tf:5]:
```suggestion
      version = "3.19.1"
```

**Comment by sebader** [comment]
> TF azurerm 3.19 was already superseeded by 3.19.1 which contains some dns resource related fixes. https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.19.1

good point. updated it

**Pull request merged**